### DISCUSSION TITLE
Add <script setup> option inside 'quasar new' for non-TS projects

### DISCUSSION THREAD

**Discussion by Random-nobody** [quasarframework/quasar#14007] [Ideas / Proposals]
Regarding comment on PR [#13994](https://github.com/quasarframework/quasar/pull/13994#issuecomment-1190398521) I would like to discuss implementing similar `<script setup>` syntax for `quasar new` command but for non-TS projects.

I'm not sure if `quasar new` is such a big deal to care about small details but after reading the source code couple question came into my mind. So alongside mentioned addition I would like to know what do you think about couple refactors alongside it. Please tell me if I'm drilling too much into details and it does not matter or my train of thought goes wrong way etc.

Currently this are supported format options (on `dev` branch):
```
--format -f <option>  (optional) Use a supported format for the template
                      Possible values:
                          * default - Default JS template
                          * ts-composition - TS composition API (default if using TS)
                          * ts-composition-setup - TS composition API with <script setup>
                          * ts-options - TS options API
                          * ts-class - [DEPRECATED] TS class style syntax
                          * ts - Plain TS template (for boot, store, and ssrmiddleware files)
```

It seems to me that it is a little bit inconsistent, as TS supports both options API and composition API, while JS have only one default preset. 
I would like to know your opinion on:

1. adding new option `js-composition-setup`
2. renaming `default` option to actual name of what it is generating (currently it is generating JS composition API so maybe `js-composition`)
3. adding `js` option for _boot, store, and ssrmiddleware files_ in simillar fasion to `ts`
4. adding new `js-options` format to make it consistent with `ts-options`

This are changes made mostly inside the source code. As for cosmetic changes for an end user we could then do:

1. if we assume that TS project will never use JS templates and vice versa maybe then we could skip `js-` and `ts-` part inside command line parameter and we would be left with `composition`, `composition-setup` and `options`. Of course inside source code folders would still have this prefixes
2. if we assume Composition APi is the default one maybe we could omit 'composition' part and we would get just `<no format provided>`, `setup`, `options`

I think this changes would make code base more readable and concise. But on the other hand renaming stuff maybe is considered breaking change? Or at the end of the day maybe we should not spend time on something like this.

I personally would prefer to just focus on composition API and slowly fade out options API. Also I expect usage of JS files slowly drop, so maybe we should not care about them that much. So in minimalist approach I would just stick to main point of adding `js-composition-setup`, even if it could make more convoluted logic inside [app-vite/lib/cmd/new.js](https://github.com/quasarframework/quasar/blob/0365bded2c0e83c2a990b6c35abef2adf5793e57/app-vite/lib/cmd/new.js) (which is more catered towards different TS formats rather than JS).

**Comment by yusufkandemir** [comment]
We can keep `default` as an alias to `js-composition`, just as a compatibility layer of some sort. A simple if check when processing the parameters would work.

Dropping the `js-`, `ts-` prefixes with that assumption could be useful. But, I think we should still keep it as an alias. So, for a TS project `composition` would point to `ts-composition`, but they can still write `js-composition` to use the JS variant. This way, we would make it easier for most users, while still making it possible for possible rare cases and avoiding introducing breaking changes.

Instead of dropping the `composition` from the format names completely, again, we can add `setup` as an alias. So, both `composition-setup` and `setup` would do the same thing.

If added carefully, it shouldn't mess up the logic too much. In the end, we are going to end up with `js`, `js-{FORMAT}`, `ts`, and `ts-{FORMAT}` formats and folders, so it's just adding the same set of logic for JS to match with the TS behavior and structure.

So, your observations on what to add and what to change are correct. But, as you noticed, some of them might end up being a breaking change, which we can avoid by using aliases as I mentioned earlier.
